### ISSUE TITLE
Ability to restrict users to SSO login only

### ISSUE THREAD

**Issue by joesaunderson** [PostHog/posthog#7535]
## Is your feature request related to a problem?

We manage around 20 3rd party tools that our internal users can log into using G Suite (part of your enterprise / scale offering). When a user leaves, we need to disable their account. Currently we need to do this in G Suite and then any of the 3rd party tools where password login is still possible.

## Describe the solution you'd like

When a user leaves, we disable their account in G Suite and their account in PostHog becomes disabled too. Essentially, the ability to restrict specific users in PostHog to only be able to login via SSO (defaulting to this setting).

## Describe alternatives you've considered

Manually removing their account, and being the enemy of our IT manager.

**Comment by macobo** [comment]
cc @paolodamico - would make for a great collaboration-related feature :)

**Comment by paolodamico** [comment]
Indeed! On Cloud it's a bit trickier because we would also need to restrict the domains that are allowed to access a certain org. We would also need to consider the user lifecycle to remove the old user account. Related to #5647.

**Comment by stuartharbr** [comment]
This is an essential feature for us. From a security perspective there's no point having SSO if the user can just bypass it with a local username and password.

Also, it's not an "Enterprise" feature - small companies want to be secure as well! :-)

**Comment by peterjgrainger** [comment]
@paolodamico I think the restriction part is already in place? Also if we removed the user lifecycle restriction would that make it simpler? I can have a go a PR if you think this is a good feature. We could make this an optional flag to solve the issue with the cloud deployment?

**Comment by peterjgrainger** [comment]
Also @paolodamico why is this an enterprise feature?

**Comment by paolodamico** [comment]
Hey @peterjgrainger! I don't think this is actually solved yet because we only offer login enforcement for SAML (not social auth like Google/GitHub/GitLab). Further we don't yet support this on PostHog Cloud as there's a significant amount of edge cases. Here's a draft proposal for how this could look like technically (please share any feedback 🙏), https://github.com/PostHog/meta/pull/40

It's an enterprise feature because it's something that generally enterprises require (strong compliance controls). In case you're curious here's our current internal guideline for what's free/paid (https://posthog.com/handbook/product/paid-features#when-should-a-feature-be-paid-or-free#when-should-a-feature-be-paid-or-free).
